### PR TITLE
Add check-client section to dg and add AddressBookParser test

### PR DESCRIPTION
**Pull request by Meowloid**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -209,9 +209,50 @@ The following activity diagram summarizes what happens when a user executes a `v
...
+The **Check Client** mechanism is facilitated by `CheckClientCommand` extending the `Command` family of classes, as elaborated in the Logic component of MATER. Additionally, it requires the following operations:
+
+* `CheckClientCommandParser#parse()` - Given the argument succeeding the `check-client` command, parses the appropriate index to `CheckClientCommand`.
+* `CheckClientCommand#execute()` - Given the list of Clients, identifies the indexed Client and toggles the "checked (in or out)" status of the Client's car.
+
+Given below is an example usage scenario and details on how the **Check Client** mechanism behaves at each step.
+
+**Step 1**: The user launches the application, where all clients are listed by default.
+
+**Step 2**: The user executes `check-client 1` command to toggle the "checked-in" status of the car for the first Client in the list.
```
Comment (xhamyo) [inline docs/DeveloperGuide.md:225]: I think should have more consistency, above is "checked (in or out)" status, here is only "checked-in" status.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -209,9 +209,50 @@ The following activity diagram summarizes what happens when a user executes a `v
...
+
+The following activity diagram summarizes what happens when a user executes a `check-client` command:
+
+<puml src="diagrams/CheckClientActivityDiagram.puml" width="250" />
+
+#### Design considerations:
+
+**Aspect: How Check Client executes:**
+
+* **Alternative 1 (current choice):** Toggles the "checked-in" status of the car for the given Client and displays a success or error message.
```
Comment (xhamyo) [inline docs/DeveloperGuide.md:249]: Same as first comment, about consistency.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -209,9 +209,50 @@ The following activity diagram summarizes what happens when a user executes a `v
...
+    * Pros: Easy to implement and understand. The success message clearly indicates whether a Client’s car has been checked in or out.
+    * Cons: Does not allow for more detailed tracking or history of each check-in/out event for a Client's car.
+
+* **Alternative 2:** Records every check-in/out event with a timestamp and additional information about the user who performed the action.
```
Comment (xhamyo) [inline docs/DeveloperGuide.md:253]: `and additional information about the user who performed the action` is not needed, I think the product is intended for one user.

**Review by xhamyo — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -209,9 +209,50 @@ The following activity diagram summarizes what happens when a user executes a `v
...
+
+**Step 1**: The user launches the application, where all clients are listed by default.
+
+**Step 2**: The user executes `check-client 1` command to toggle the "checked-in" status of the car for the first Client in the list.
+
+**Step 3**: If the Client has a car associated, their car's "checked-in" status is toggled, and the success message indicating the action is displayed. If the Client does not have a car associated, an error message indicating there is **"No Car associated to Client to Check In"** is displayed.
+
+**Step 4**: The user may check-in/out the status of other clients by providing the relevant index of the client they wish to edit the status of.
+
+<box type="info" seamless>
```
[the remark this answers is not in the document]
Comment (Meowloid) [reply docs/DeveloperGuide.md:231]: Noted

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -209,9 +209,50 @@ The following activity diagram summarizes what happens when a user executes a `v
...
+
+</box>
+
+The following sequence diagram shows how a `check-client` operation goes through the `Logic` component:
+
+<puml src="diagrams/CheckClientSequenceDiagram.puml" alt="CheckClientSequenceDiagram" />
+
```
[the remark this answers is not in the document]
Comment (Meowloid) [reply docs/DeveloperGuide.md:238]: Done

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Meowloid) [reply docs/DeveloperGuide.md:253]: Noted

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 6]
```diff
@@ -99,6 +100,12 @@ public void parseCommand_list() throws Exception {
         assertTrue(parser.parseCommand(ListClientCommand.COMMAND_WORD + " 3") instanceof ListClientCommand);
     }
 
+    @Test
+    public void parseCommand_addCar() throws Exception {
+        assertTrue(parser.parseCommand(AddCarCommand.COMMAND_WORD + " 2 vrn/SGX1234B "
+                + "vin/KMHGH4JH3EU073801 make/Crazy model/Taxi") instanceof AddCarCommand);
+    }
+
```
Comment (SeanWong2) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:108]: Is the make and model correct?

Comment (Meowloid) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:108]: if the shoe fits...

**Pull request merged**